tiledb/cloud/pool_manager_wrapper.py

```diff
@@ -1,17 +1,41 @@
+import urllib
+
+
 class _PoolManagerWrapper:
     """
     Wrapper around urllib3.PoolManager that implements request() which
     is the only method used by the autogenerated API code
 
     It can be used to replace tiledb.cloud.rest_api.rest.RESTClientObject.pool_manager
-    at runtime to work around https://github.com/urllib3/urllib3/issues/2475
+    at runtime in order to cache HTTP redirects and work around
+    https://github.com/urllib3/urllib3/issues/2475
     """
 
     def __init__(self, pool_manager):
         self._pool = pool_manager
+        self._redirect_cache = {}
 
     def request(self, method, url, **kwargs):
         kwargs["retries"] = self._pool.connection_pool_kw.get("retries")
+
+        parsed_url = urllib.parse.urlparse(url)
+        cacheable_url = urllib.parse.urljoin(
+            f"{parsed_url.scheme}://{parsed_url.netloc}",
+            parsed_url.path,
+        )
+
+        cached_netloc = self._redirect_cache.get(cacheable_url)
+        if cached_netloc:
+            url = parsed_url._replace(netloc=cached_netloc).geturl()
+
         resp = self._pool.request(method, url, **kwargs)
 
+        for retry in reversed(resp.retries.history):
+            if retry.redirect_location:
+                self._redirect_cache[cacheable_url] = urllib.parse.urlparse(
```

Comment:
This will probably be easier to read if it's broken into two parts:

```python
dest_domain = urllib.parse.urlparse(retry.redirect_location).netloc
self._redirect_cache[cacheable_url] = dest_domain
```

Reply:
done